app/database/queries/breeze_buddy/chat_session.py

```
@@ -107,6 +107,28 @@ def end_chat_session_query(
+    ``update_outcome`` builtin that fires it) so a chat / assist session
+    records an outcome as soon as the LLM sets one — instead of dropping
+    it, which is what happened before when the hook bailed on
+    ``context.lead`` being None. Returns the written outcome.
+    """
+    query = f"""
+        UPDATE {CHAT_SESSION_TABLE}
+        SET outcome = $2,
+            last_activity_at = now()
+        WHERE id = $1
+        RETURNING outcome
+    """
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)